tin/apps/submissions/models.py

```diff
@@ -27,14 +27,23 @@ def filter_visible(self, user):
             return self.all()
         else:
             return self.filter(
-                Q(assignment__course__teacher=user) | Q(student=user) & Q(assignment__is_quiz=False)
+                Q(assignment__course__teacher=user)
+                | Q(student=user)
+                & Q(assignment__is_quiz=False)
+                & (
+                    Q(assignment__course__archived=False)
+                    | Q(assignment__course__permission__in="rw")
+                )
             ).distinct()
 
     def filter_editable(self, user):
         if user.is_superuser:
             return self.all()
         else:
-            return self.filter(Q(assignment__course__teacher=user) | Q(student=user)).distinct()
+            return self.filter(
+                Q(assignment__course__teacher=user)
+                | (Q(student=user) & Q(assignment__course__archived=False))
+            ).distinct()
```

Comment:
I know this isn't really a part of the PR, but I wonder whether students should even be allowed to "edit" submissions.

Taking a look at the code, seems like `filter_editable()` is used in 6 actions:
- killing a submission
- adding, editing, and deleting comments
- publishing and unpublishing the submission

Feels to me like students should only be able to do the first one (killing their submission), and this is reflected in the UI as well. So, wouldn't it make more sense to remove students from this permission and change kill_view to use `filter_visible()`?

Reply:
That makes sense to me, I was wondering the same thing initially lol
Let me add some tests for `kill_view`, and then we should be good :)